KafkaIngester/src/main/java/io/deephaven/kafka/ingest/GenericRecordConsumerRecordToTableWriterAdapter.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright (c) 2016-2020 Deephaven Data Labs and Patent Pending
+ */
+
+package io.deephaven.kafka.ingest;
+
+import io.deephaven.tablelogger.RowSetter;
+import io.deephaven.tablelogger.TableWriter;
+import io.deephaven.db.tables.utils.DBDateTime;
+import io.deephaven.db.tables.utils.DBTimeUtils;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.avro.generic.GenericRecord;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ *  Convert an Avro {@link GenericRecord} to Deephaven rows.
+ *
+ *  Each GenericRecord produces a single row of output, according to the maps of Table column names to Avro field names
+ *  for the keys and values.
+ */
+public class GenericRecordConsumerRecordToTableWriterAdapter implements ConsumerRecordToTableWriterAdapter {
+    private final TableWriter<?> writer;
+    private final Map<String, String> keyColumns;
+    private final Map<String, String> valueColumns;
+    private final RowSetter<Integer> kafkaPartitionColumnSetter;
+    private final RowSetter<Long> offsetColumnSetter;
+    private final RowSetter<DBDateTime> timestampColumnSetter;
+    @SuppressWarnings("rawtypes")
+    private final RowSetter rawKeyColumnSetter;
+    private final Map<String, Function<GenericRecord, ?>> columnToKeyFunction;
+    private final Map<String, Function<GenericRecord, ?>> columnToValueFunction;
+    private final boolean allowNullKeys;
+    private final boolean allowNullValues;
+
+    private GenericRecordConsumerRecordToTableWriterAdapter(
+            final TableWriter<?> writer,
+            final String kafkaPartitionColumn,
+            final String offsetColumnName,
+            final String timestampColumnName,
+            String rawKeyColumnName,
```

Comment:
This is effectively subsuming simple for the key column; but doing generic records for the value column.

Reply:
Lost, I don't understand your comment.  Feel free to steer me in the direction of the actual change/fix here.
Keep in mind tho that these classes will go away, we won't do TableWriter based, correct?

Reply:
It's more of a generic comment that we want to split the ConsumerRecordToStreamPublisher adapters by "generic stuff", "key stuff" and "value stuff".  Pointing out that primitives keys are important enough to have them built into the generic record adapter.